lib/util/glob-util.js

```diff
@@ -78,11 +78,15 @@ function processPath(options) {
 
 /**
  * Resolves any directory patterns into glob-based patterns for easier handling.
- * @param   {string[]} patterns    File patterns (such as passed on the command line).
- * @param   {Object} options       An options object.
+ * @param   {string[]} patterns               File patterns (such as passed on the command line).
+ * @param   {Object} options                  An options object.
+ * @param   {string} [options.globInputPaths] False disables glob resolution.
  * @returns {string[]} The equivalent glob patterns and filepath strings.
  */
 function resolveFileGlobPatterns(patterns, options) {
+    if (options.globInputPaths === false) {
```

Comment:
We could test `!options.globInputPaths` here indeed as `globInputPaths` would be set by default to `true`. However, the default setting (`true`) is set higher up in the calls, so a lot of unit tests would need to be changed to add `globInputPaths: true`.

Also there is a similar situation in the same file [here](https://github.com/pvdlg/eslint/blob/cecfa4cf3fe8c9f4c2c74d1cb8a7956cd635dd0a/lib/util/glob-util.js#L143) where the `ignore` option is explicitly compared to `false` even though it's [defaulted to `true`](https://github.com/pvdlg/eslint/blob/cecfa4cf3fe8c9f4c2c74d1cb8a7956cd635dd0a/conf/default-cli-options.js#L16).

In the end I favored consistency and simplicity so I went with `=== false`. But I don't have a preference either way. I can do whatever you prefer.

Reply:
I think it would be a good idea to change these eventually so that the implementation is simpler (we don't need to deal with defaults multiple times), but since this pattern is already in the file, I think it's fine to keep it as-is in this PR and maybe we can change everything later.